### ISSUE TITLE
🐛 Parser rejects a in b ? x : { … (p) => … } (arrow with parenthesized params in ternary alternate after an in test); formatter turns valid input into this failing form

### ISSUE THREAD

**Issue by repetitioestmaterstudiorum** [biomejs/biome#11464]
### Environment information

Environment: @biomejs/biome 2.2.0 and 2.5.7 (TypeScript parses all variants; the 2.5.8–2.5.10 changelog lists no related JS parser fix).

Reproduce: `biome check repro.ts`

repro.ts:
```ts
declare function tool(x: { f: (o: { output: { error: string } | { items: { k: string }[] } }) => unknown }): unknown;
export const t = tool({
  f: ({ output }) => ({ v: "error" in output ? output : { items: output.items.map((i) => ({ k: i.k })) } }),
});
```

Output:
```
repro.ts:3:28 parse  × Expected an identifier, an array pattern, or an object pattern but instead found '"error"'.
repro.ts:3:36 parse  × expected `:` but instead found `in`
repro.ts:3:39 parse  × expected `:` but instead found `output`
repro.ts:3:46 parse  × expected `,` but instead found `?`
repro.ts:3:48 parse  × expected `,` but instead found `output`
repro.ts:3:82 parse  × ';' expected
```
The outer parenthesized arrow body `({ v: … })` is re-parsed as an arrow parameter list.

Trigger matrix (each change alone makes it parse fine):
- arrow function in the consequent instead of the alternate ✔
- `in` replaced by `Object.hasOwn(...)` ✔
- nested `function` expression instead of an arrow ✔
- unparenthesized nested param `i => …` ✔
- parenthesized test `("error" in output) ? …` ✔
- outer block body instead of `=> ({ … })` ✔

What's kind of funny/sad: The formatter strips the parentheses around the `in` test and re-adds parentheses around the nested arrow's parameter (`arrowParentheses: "always"`), so `biome format` rewrites both passing variants into the failing one, so valid input becomes unparseable output.

Possibly related to the ternary-consequent arrow handling in #1077 and #10131 / #10152.

### What happened?

1. Create `repro.ts` with exactly this content (valid TypeScript; `tsc` accepts it):

```ts
declare function tool(x: { f: (o: { output: { error: string } | { items: { k: string }[] } }) => unknown }): unknown;
export const t = tool({
  f: ({ output }) => ({ v: "error" in output ? output : { items: output.items.map((i) => ({ k: i.k })) } }),
});
```

2. Run `biome check repro.ts` (reproduced on 2.2.0 and 2.5.7; the 2.5.8–2.5.10 changelog lists no related JS parser fix).

3. Biome fails to parse the file:

```
repro.ts:3:28 parse  × Expected an identifier, an array pattern, or an object pattern but instead found '"error"'.
repro.ts:3:36 parse  × expected `:` but instead found `in`
repro.ts:3:39 parse  × expected `:` but instead found `output`
repro.ts:3:46 parse  × expected `,` but instead found `?`
repro.ts:3:48 parse  × expected `,` but instead found `output`
repro.ts:3:82 parse  × ';' expected
```
The outer parenthesized arrow body `({ v: … })` is re-parsed as an arrow parameter list.

Trigger matrix (each change alone makes it parse fine):
- arrow function in the consequent instead of the alternate ✔
- `in` replaced by `Object.hasOwn(...)` ✔
- nested `function` expression instead of an arrow ✔
- unparenthesized nested param `i => …` ✔
- parenthesized test `("error" in output) ? …` ✔
- outer block body instead of `=> ({ … })` ✔

4. The two style-level escapes don't survive formatting: `biome format` strips the parentheses around the `in` test and re-adds parentheses around the nested arrow's parameter (`arrowParentheses: "always"`), rewriting both passing variants into the failing one — so valid input becomes output Biome itself can't parse.

Possibly related to the ternary-consequent arrow handling in #1077 and #10131 / #10152.

### Expected result

`biome check repro.ts` reports no diagnostics: the file is valid TypeScript (`a in b ? x : { … (p) => … }` inside a parenthesized arrow body), and `biome format` must never rewrite parseable input into output that Biome's own parser rejects.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
playground: https://biomejs.dev/playground/?lintRules=none&tab=formatter&pane=Diagnostics#code=ZABlAGMAbABhAHIAZQAgAGYAdQBuAGMAdABpAG8AbgAgAHQAbwBvAGwAKAB4ADoAIAB7ACAAZgA6ACAAKABvADoAIAB7ACAAbwB1AHQAcAB1AHQAOgAgAHsAIABlAHIAcgBvAHIAOgAgAHMAdAByAGkAbgBnACAAfQAgAHwAIAB7ACAAaQB0AGUAbQBzADoAIAB7ACAAawA6ACAAcwB0AHIAaQBuAGcAIAB9AFsAXQAgAH0AIAB9ACkAIAA9AD4AIAB1AG4AawBuAG8AdwBuACAAfQApADoAIAB1AG4AawBuAG8AdwBuADsACgBlAHgAcABvAHIAdAAgAGMAbwBuAHMAdAAgAHQAIAA9ACAAdABvAG8AbAAoAHsACgAgACAAZgA6ACAAKAB7ACAAbwB1AHQAcAB1AHQAIAB9ACkAIAA9AD4AIAAoAHsAIAB2ADoAIAAiAGUAcgByAG8AcgAiACAAaQBuACAAbwB1AHQAcAB1AHQAIAA%2FACAAbwB1AHQAcAB1AHQAIAA6ACAAewAgAGkAdABlAG0AcwA6ACAAbwB1AHQAcAB1AHQALgBpAHQAZQBtAHMALgBtAGEAcAAoACgAaQApACAAPQA%2BACAAKAB7ACAAawA6ACAAaQAuAGsAIAB9ACkAKQAgAH0AIAB9ACkALAAKAH0AKQA7AA%3D%3D

Please include the playground link next time. And for a bug report like this, a typescript playground link would also be useful.

Is this the smallest reproduction you have?

**Comment by repetitioestmaterstudiorum** [comment]
Thanks. The first example was more or less as in my code; here is a much smaller one. Reducing it also showed that `in` is not the trigger at all, so the title is now wrong.

```ts
const apply = (callback: (n: number) => number) => callback(1);
const fn = (flag: boolean) => ({ value: flag ? flag : { result: apply((n) => n) } });
```

Biome playground: https://biomejs.dev/playground/?lintRules=none&tab=formatter&pane=Diagnostics#code=YwBvAG4AcwB0ACAAYQBwAHAAbAB5ACAAPQAgACgAYwBhAGwAbABiAGEAYwBrADoAIAAoAG4AOgAgAG4AdQBtAGIAZQByACkAIAA9AD4AIABuAHUAbQBiAGUAcgApACAAPQA%2BACAAYwBhAGwAbABiAGEAYwBrACgAMQApADsACgBjAG8AbgBzAHQAIABmAG4AIAA9ACAAKABmAGwAYQBnADoAIABiAG8AbwBsAGUAYQBuACkAIAA9AD4AIAAoAHsAIAB2AGEAbAB1AGUAOgAgAGYAbABhAGcAIAA%2FACAAZgBsAGEAZwAgADoAIAB7ACAAcgBlAHMAdQBsAHQAOgAgAGEAcABwAGwAeQAoACgAbgApACAAPQA%2BACAAbgApACAAfQAgAH0AKQA7AAoA

TS playground: https://www.typescriptlang.org/play/?ts=6.0.3#code/MYewdgzgLgBAhgBwQGwJ4wLwwBTDs5AIzmAGsAuHMSsAVwFtCBTAJwEpMA+GOx1jjNzwFiZbAEY2AbgBQoSLABmYTDkXI4Ac0qEQIZEzhgB3bAG8YAN3y0mldVpgB+GA80xKFlkwi1kUSkQUVGxsYy4eDgBfGCjpGSA

**Comment by dyc3** [comment]
Honestly, I'm surprised this hadn't been reported earlier

**Comment by repetitioestmaterstudiorum** [comment]
Yeah a little bit.
Thank you for the solution. Impressive speed!
I appreciate your work on biome. Have a good weekend.
